metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiling/partition_discovery.py

```diff
@@ -0,0 +1,2250 @@
+                return self._remove_duplicate_columns(function_columns)
+
+            # If no columns found, return empty list
+            logger.debug(
+                f"No column names extracted from partition clause: {partition_clause}"
+            )
+            return []
+
+        except Exception as e:
+            logger.warning(
+                f"Error extracting column names from partition clause '{partition_clause}': {e}"
+            )
+            return []
+
+    def _extract_simple_column_names(self, partition_clause: str) -> List[str]:
+        """Extract simple comma-separated column names from partition clause."""
+        # Pattern for simple column names (col1, col2, col3)
+        simple_pattern = r"^([a-zA-Z_][a-zA-Z0-9_]*(?:\s*,\s*[a-zA-Z_][a-zA-Z0-9_]*)*)$"
+        simple_match = re.match(simple_pattern, partition_clause.strip())
+
+        if simple_match:
+            # Direct column names separated by commas
+            columns = [col.strip() for col in simple_match.group(1).split(",")]
+            return [col for col in columns if col]
+
+        return []
+
+    def _extract_function_based_column_names(self, partition_clause: str) -> List[str]:
+        """Extract column names from function-based partitioning patterns."""
+        column_names = []
+        function_patterns = self._get_function_patterns()
+        seen = set()
+
+        for pattern in function_patterns:
+            matches = re.findall(pattern, partition_clause, re.IGNORECASE)
+            for match in matches:
+                if match not in seen:
+                    seen.add(match)
+                    column_names.append(match)
+
+        return column_names
+
+    def _extract_mixed_column_names(self, partition_clause: str) -> List[str]:
+        """Extract column names from mixed scenarios (comma-separated parts with functions)."""
+        column_names = []
+        function_patterns = self._get_function_patterns()
+
+        # Use smarter splitting that respects parentheses
+        parts = self._split_partition_clause_respecting_parentheses(partition_clause)
+
+        for part in parts:
+            part = part.strip()
+            found_match = False
+
+            # Try to extract column name from each part using function patterns
+            for pattern in function_patterns:
+                matches = re.findall(pattern, part, re.IGNORECASE)
+                if matches:
+                    column_names.extend(matches)
+                    found_match = True
+                    break  # Stop after first successful pattern match to avoid duplicates
+
+            # If no function found, check if it's a simple column name
+            if not found_match:
+                simple_col_match = re.match(r"^([a-zA-Z_][a-zA-Z0-9_]*)$", part)
+                if simple_col_match:
+                    column_names.append(simple_col_match.group(1))
+
+        return column_names
+
+    def _get_function_patterns(self) -> List[str]:
```

Comment:
These should be compiled regexp. I still prefer using sql parser vs parsing by hand which is not very reliable